cairo-libs/packages/protobuf/src/tests/proto.cairo

```diff
@@ -55,14 +58,17 @@ impl ValidatorAsProtoMessage of ProtoMessage<ValidatorType> {
         }
     }
 
-    fn decode_raw(ref self: ValidatorType, ref context: DecodeContext) {
-        let mut var = Default::<u32>::default();
-        var.decode_raw(ref context);
-        match var {
-            0 => self = ValidatorType::Full,
-            1 => self = ValidatorType::Light,
-            _ => panic!("invalid validator type"),
-        }
+    fn decode_raw(ref context: DecodeContext) -> Option<ValidatorType> {
+        let var: Option<u32> = context.decode_raw();
+        if var.is_none() {
+            return Option::None;
+        };
```

Comment:
Careful with `if _.is_none() { return None; }` blocks. They can be simplified with `?` operator.

Reply:
[3e888ee](https://github.com/informalsystems/ibc-starknet/pull/281/commits/3e888eea10b8b78e9fd0586959f2db2000be483b)
By the way, note that `?` is not allowed inside loops. So I've left them as is.